libs/experimental/langchain_experimental/openai_assistant/base.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import annotations
+
+import json
+from time import sleep
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Sequence, Union
+
+from langchain.pydantic_v1 import Field, root_validator
+from langchain.schema.agent import AgentAction, AgentFinish
+from langchain.schema.runnable import RunnableConfig, RunnableSerializable
+from langchain.tools import format_tool_to_openai_function
+from langchain.tools.base import BaseTool
+
+if TYPE_CHECKING:
+    import openai
+
+
+class OpenAIAssistantFinish(AgentFinish):
+    run_id: str
+    thread_id: str
+
+
+class OpenAIAssistantAction(AgentAction):
+    tool_call_id: str
+    run_id: str
```

Comment:
Could you document why run_id is here? (I assume just for debugging purposes -- i.e., if someone needs to access it to see what's going on?)

Reply:
no needed for agent with custom tools, will document